_posts/2024/2024-02-06-our-process.md

```diff
@@ -0,0 +1,453 @@
+
+-   Summary: Ongoing assessment of engineering's needs
+-   Key Questions:
+    -   What parts of our tools, systems, architecture are slowing us down?
+    -   What new technical advancements should we be aware of?
+    -   What do we *need* vs. what do we *want*?
+-   Frequency: ongoing
+-   Responsible: technical leads
+-   Contributors: entire engineering team
+-   Products:
+    -   Designs for new features or applications
+
+### Biweekly Sprint Goals/Backlog Grooming
+
+OK,
+we've agreed on what we're going to build—now we actually have to build it.
+Every two weeks,
+the Scrum master and the project's technical lead
+create issues for new work
+and (if necessary) relabel old issues
+to define what we're going to try to accomplish in this sprint.
+Most of the new issues come from the design docs,
+or are elaborations or breakdowns of things discussed in those docs;
+depending on the size of the work,
+we may create several sub-issues under a single high-level issue
+so that progress reports aren't "I'm working on it" several days in a row.
+
+-   Summary: Biweekly prioritization to determine team focus for the sprint
+-   Key Questions:
+    -   What is the next slice of work we want to deliver?
+    -   What small improvements should we prioritize?
+    -   What key bugs should we prioritize?
+    -   What tech debt needs to be addressed?
+-   Frequency: biweekly
+-   Responsible: Scrum master and technical leads
+-   Contributors: engineering team members
+-   Products:
+    -   Labeled issues
+
+> I've worked on teams where issues had a "percent complete" field
+> that the people responsible for it were supposed to update daily.
+> It wasn't helpful:
+> it's easy to see how much ditch you have left to dig,
+> but much harder to know how much code you have left to write and debug.
+> Breaking milestones down into inch-pebbles
+> and marking the latter as "finished" or "not finished"
+> has proven much more useful.
+
+### Daily Standups
+
+We don't actually stand up,
+but this is still the most important meeting each team has.
+The tech lead of the project does a quick lap around the team to find out
+what each person has done,
+what they're planning to do,
+and what's holding them back.
+We try to enforce a three-sentence rule:
+if a discussion during individual updates goes on longer than that
+we defer it until after everyone has finished their initial report.
+
+The "output" from standups is course corrections.
+Should we try another approach to a problem?
+Did someone discover an issue that ought to go in the backlog for the next sprint,
+or something that's important enough to replace something planned for this sprint?
+
+-   Summary: Daily team sync
+-   Key Questions:
+    -   What did you work on yesterday?
+    -   What are you working on today?
+    -   What is blocking you?
+-   Frequency: daily
+-   Responsible: technical leads
+-   Contributors: engineering team members
+-   Products:
+    -   Updated issues
```

Comment:
Perhaps it is helpful to be more specific here?
- Progress update on prioritized work
- Reprioritized issues and goals, as necessary

Reply:
updated